src/tallies/MeshTally.C

```diff
@@ -121,12 +121,6 @@ MeshTally::MeshTally(const InputParameters & parameters)
           paramError("blocks",
                      "Block '" + block_names[b] + "' specified in 'blocks' not found in mesh!");
     }
-    else
```

Comment:
Do we not want to keep this default behavior? If there are a lot of blocks, is this requiring the user to type out all the block names?

Reply:
I changed how `MeshTally` performs the block check so filling `_tally_blocks` is no longer required when tallying over all blocks. Instead, `MeshTally` checks if `_tally_blocks` is empty, and if so the full mesh is used to initialize the mesh tally.